### PR TITLE
feat: form feature

### PR DESCRIPTION
**Pull request by ankit9015**
Form feature includes responsiveness according to:
1. Valid 
2. Invalid
3. Focus
4. Required
5. Empty input (:placeholder-shown)

https://user-images.githubusercontent.com/67504902/154498568-f836bf14-8381-41bf-bf36-7bb9e79b7605.mp4

### CODE REVIEW

File: components/form/form.css [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    border-color: var(--focus-border);
+    opacity: 100%;
+}
+
+.socketui-input:valid {
+    border-color: var(--input-valid);
+    opacity: 100%;
+}
+
+
```
Comment (Kothalkarpriya) [inline components/form/form.css:27]: remove extra white spaces

File: components/form/form.css [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+.socketui-input:placeholder-shown {
+    border-color: var(--input-border);
+}
+
+.socketui-input:focus:invalid {
+    border-color: var(--input-invalid);
+    background-color: var(--error__bg-color);
+}
+
+
```
Comment (Kothalkarpriya) [inline components/form/form.css:49]: here also

File: components/form/form.css [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+}
+
+.label-required::after {
+    content: '*';
+    color: red;
+    font-size: 1rem;
+    margin: 1ch;
+}
+
+
```
Comment (Kothalkarpriya) [inline components/form/form.css:77]: here also

**Review by Kothalkarpriya — APPROVED** [review]
Look into comments, then you are good to go

File: components/badge/badge.css [hunk 4]
```diff
@@ -25,3 +25,10 @@
     flex-direction: row;
     justify-content: space-around;
 }
+
+.badge-text {
+    font-size: 1em;
+    padding: 0.1rem 0.35rem;
+    border-radius: 0.05em;
+    position: absolute;
+}
```
Comment (sks-21) [inline components/badge/badge.css:34]: for values less than 10px, don't use rem. Prefer pixels

**Review by sks-21 — APPROVED** [review]
The code looks nice. Good to merge.

**Pull request merged**